### PR TITLE
Fix Notification ObjectLock

### PR DESCRIPTION
**Pull request by tsmr**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | ###

When you define a notification for  ObjectLock, the notficiation is not sended because it's not the good id of the ticket which beeing sended

### CODE REVIEW

File: src/ObjectLock.php [hunk 1]
```diff
@@ -247,7 +247,7 @@ function askUnlock() {
                      cache: false,
                      data: {
                         requestunlock: 1,
-                        id: {$this->fields['id']}
+                        id: {$this->itemid}
```
Comment (cedric-anne) [inline src/ObjectLock.php:250]: The `requestunlock` action will try to load the `ObjectLock` instance using `(new ObjectLock())->getFromDB($_POST["id"])`, so, unless I miss something, the you have to pass `$this->fields['id']` to the action.

Comment (tsmr) [reply src/ObjectLock.php:250]: When you use id: {$this->fields['id']} the sended id is 1 and not the ticket's id

Comment (tsmr) [reply src/ObjectLock.php]: You're right :)
Better fix :P

**Review by cedric-anne — DISMISSED** [review]

File: src/ObjectLock.php (hunk 1 above)
Comment (tsmr) [reply src/ObjectLock.php]: You have this error if you not do this : 

```
[2023-11-10 07:27:20] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "date_mod" in C:\wamp64\www\glpi10\src\NotificationTargetObjectLock.php at line 122
  Backtrace :
  src\NotificationTarget.php:1363                    NotificationTargetObjectLock->addDataForTemplate()
  src\NotificationTemplate.php:278                   NotificationTarget->getForTemplate()
  src\NotificationEventAbstract.php:122              NotificationTemplate->getTemplateByLanguage()
  src\NotificationEvent.php:187                      NotificationEventAbstract::raise()
  ajax\unlockobject.php:78                           NotificationEvent::raiseEvent()
```

**Comment by cedric-anne** [comment]
This PR seems stale, I close it. Feel free to reopen it.

**Pull request closed without merging**